From 46a8044450ee07e917f7a3315e46fab0e986fa49 Mon Sep 17 00:00:00 2001
From: Kalyan Kondapally <kalyan.kondapally@intel.com>
Date: Wed, 10 Aug 2016 15:26:15 -0700
Subject: [PATCH 1/3] Hack: Avoid going to sleep mode.

We dont have proper support for system restore and this is cauing
the system reboot as soon as we go to low power state. This is really
annoying on my Dell XPS laptop where reboot happens after 1-2 mins.
---
 .../core/java/com/android/server/power/PowerManagerService.java     | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/power/PowerManagerService.java b/services/core/java/com/android/server/power/PowerManagerService.java
index 2824e6e..96ce867 100644
--- a/services/core/java/com/android/server/power/PowerManagerService.java
+++ b/services/core/java/com/android/server/power/PowerManagerService.java
@@ -1248,7 +1248,9 @@ public final class PowerManagerService extends SystemService
                     + ", reason=" + reason + ", flags=" + flags + ", uid=" + uid);
         }

-        if (eventTime < mLastWakeTime
+        return false;
+
+        /*if (eventTime < mLastWakeTime
                 || mWakefulness == WAKEFULNESS_ASLEEP
                 || mWakefulness == WAKEFULNESS_DOZING
                 || !mBootCompleted || !mSystemReady) {
@@ -1309,7 +1311,7 @@ public final class PowerManagerService extends SystemService
         } finally {
             Trace.traceEnd(Trace.TRACE_TAG_POWER);
         }
-        return true;
+        return true;*/
     }

     private void napInternal(long eventTime, int uid) {
--
2.7.4
